apps/idos-data-dashboard/src/routes/dashboard/wallets/components/add-wallet.tsx

```diff
@@ -73,14 +75,31 @@ export const AddWallet = ({ isOpen, onClose }: AddWalletProps) => {
 
   const addWallet = useAddWalletMutation();
 
-  const handleSubmit = (event: FormEvent) => {
+  const handleSubmit = async (event: FormEvent) => {
     event.preventDefault();
 
     const form = event.currentTarget as HTMLFormElement;
     const address = new FormData(form).get("address") as string;
+    const address_regexp = /^0x[0-9a-fA-F]{40}$/;
+    const address_type = address_regexp.test(address) ? "EVM" : "NEAR";
+    let public_key: string;
+    if (address_type === "NEAR") {
+      public_key = await getNearFullAccessPublicKey(address);
+      if (!public_key) {
+        toast({
+          title: "Error while adding wallet",
+          description: "This is not correct ENV or NEAR wallet",
```

Comment:
I'd say something a bit more detailed.

```suggestion
          description: "This doesn't look like an EVM wallet, and we can't find a FullAccessKey for this NEAR address. idOS doesn't support this wallet.",
```